### DISCUSSION TITLE
Invalid request parameters

### DISCUSSION THREAD

**Discussion by rsantosdosanjos** [abuzuhri/Amazon-SP-API-CSharp#395] [General]
var messages = new List<InventoryMessage>();
            messages.Add(BuildInventory(new StockDTO(sku, string.Empty, 0, new List<WarehouseStockDTO>() { new 
            WarehouseStockDTO(509, "") }, Enum.ProductMarketplaceMapType.Premium)));

                var amazonSellingPartner = AmazonSellingPartner();

                ConstructFeedService document = new ConstructFeedService(credentials.Merchant, "1.02");

                document.AddInventoryMessage(messages);

                var xml = document.GetXML();

                return amazonSellingPartner.Feed.SubmitFeed(
                    xml,
                    FeedType.POST_INVENTORY_AVAILABILITY_DATA,
                    new List<string>() { FikaAmazonAPI.Utils.MarketPlace.Brazil.ID },
                    null,
                    ContentType.XML);
`

`{
  "errors": [
    {
      "code": "InvalidInput",
      "message": "Invalid request parameters",
      "details": ""
    }
  ]
}`

I'm getting an error when trying to update the Inventory

**Comment by rsantosdosanjos** [comment]
I noticed that in the PostXMLDataAsync method the responseStr variable returns empty, do you know what it could be?

**Reply by RenzoF** [reply]
Can you share the contents of the xml variable?

**Reply by rsantosdosanjos** [reply]
`<?xml version="1.0" encoding="utf-16"?><AmazonEnvelope xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"><Header><DocumentVersion>1.02</DocumentVersion><MerchantIdentifier>A2K9KZY06FI605</MerchantIdentifier></Header><MessageType>Inventory</MessageType><Message><MessageID>1</MessageID><OperationType>Update</OperationType><Inventory><SKU>AC530-0001</SKU><Quantity>509</Quantity></Inventory></Message></AmazonEnvelope>`

**Reply by abuzuhri** [reply]
Can you try to same for SKU without "**-**"

**Reply by rsantosdosanjos** [reply]
The error still persists even with sku without "-"

**Reply by rsantosdosanjos** [reply]
Could the problem be related to the fact that this method is returning an empty string?
![image](https://user-images.githubusercontent.com/19205139/202167572-616b53f8-9ef2-4a71-aaa7-50c48b5a1caf.png)

or could it be related to token permissions?

![image](https://user-images.githubusercontent.com/19205139/202168499-119f9586-4d2c-4ef6-b37e-b240f6c7cb56.png)

**Reply by RenzoF** [reply]
I can't recreate this on my side.
Can you submit any feed successfully?
if not my guess will be checking the credentials

**Reply by rsantosdosanjos** [reply]
Only GetItem works, but when I try to update price or stock I get this error message. I'll check the credentials!

**Reply by abuzuhri** [reply]
@rsantosdosanjos we never have this issue and i update price and qty without any problem , please reach to us in private to check what you do wrong
